File: WebClient/src/models/view-data/malfunction-view-data.ts
import { Detail } from '../entities/detail';

// класс для представления данных о неисправностях
export class MalfunctionViewData{
  public isSelected: boolean;
  // констркутор
  constructor(
    public id: number,  // id - неисправности
    public title: string, // название неисправности
    public price: number, // количество времени на устранение неисправности
    public details: Detail[] // список деталей которые необходимы для устранения неисправности
  ){
    this.isSelected = false;
  }
}

File: WebApplication/WebApplication/Models/ViewData/MalfunctionViewData.cs
using System.Collections.Generic;
using System.Linq;
using Maintenance.Models.MaintenanceEntities;

namespace WebApplication.Models.ViewData
{
    public class MalfunctionViewData {
        // id неисправности
        public int Id { get; set; }
        // название неисправности
        public string Title { get; set; }
        // время на устранение неисправности
        public int TimeToFix { get; set; }
        // список деталей для ремонта
        public List<Detail> Details { get; set; }

        public MalfunctionViewData(){}
        public MalfunctionViewData(Malfunction malfunction, List<Detail> details)
        {
            Id = malfunction.Id;
            Title = malfunction.Title;
            TimeToFix = malfunction.TimeToFix;
            Details = details;

            Price = Details.Sum(d => d.Price);
        }

        // стоимость неисправности
        public double Price { get; private set; }
    }
}
